File: examples/src/Slides/CRUD/CloneAnotherPresentationAtSpecifiedPosition.cs
﻿using Aspose.slides.Export

/*
This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Slides for .NET API reference 
when the project is build. Please check https:#docs.nuget.org/consume/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Slides for .NET API from http:#www.aspose.com/downloads, 
install it and then add its reference to this project. For any issues, questions or suggestions 
please feel free to contact us using http:#www.aspose.com/community/forums/default.aspx
*/

namespace Aspose.slides.Examples.CSharp.slides.CRUD
{
    class CloneAnotherPresentationAtSpecifiedPosition
    {
        public static void Run()
        {
            #ExStart:CloneAnotherPresentationAtSpecifiedPosition
            # The path to the documents directory.
            dataDir = RunExamples.GetDataDir_Slides_Presentations_CRUD()

            # Instantiate Presentation class to load the source presentation file
            using (Presentation sourcePresentation = new Presentation(dataDir + "AccessSlides.pptx"))
            {
                # Instantiate Presentation class for destination presentation (where slide is to be cloned)
                using (Presentation destPres = new Presentation())
                {
                    # Clone the desired slide from the source presentation to the end of the collection of slides in destination presentation
                    ISlideCollection slideCollection = destPres.Slides

                    # Clone the desired slide from the source presentation to the specified position in destination presentation
                    slideCollection.insert_clone(1, sourcePresentation.slides[1])

                    # Write the destination presentation to disk
                    destPres.save(dataDir + "CloneAnotherPresentationAtSpecifiedPosition_out.pptx", slides.export.SaveFormat.PPTX)
                }
            }
            #ExEnd:CloneAnotherPresentationAtSpecifiedPosition
        }
    }
}

File: examples/src/slides/crud/clone_another_presentation_at_specified_position.py
import aspose.slides as slides


def clone_another_presentation_at_specified_position(global_opts):
    # Instantiate Presentation class to load the source presentation file
    with slides.Presentation(global_opts.data_dir + "welcome-to-powerpoint.pptx") as source_presentation:
        # Instantiate Presentation class for destination presentation (where slide is to be cloned)
        with slides.Presentation() as destination_presentation:
            # Clone the desired slide from the source presentation to the end
            # of the collection of slides in destination presentation
            slide_collection = destination_presentation.slides

            # Clone the desired slide from the source presentation to the specified position in destination presentation
            slide_collection.insert_clone(1, source_presentation.slides[1])

            # Write the destination presentation to disk
            destination_presentation.save(global_opts.out_dir + "crud_insert_clone_out.pptx",
                                          slides.export.SaveFormat.PPTX)
